controllers/bpfman-agent/internal/test-utils/fake_bpfman_client.go

```diff
@@ -55,8 +54,12 @@ func NewBpfmanClientFakeWithPrograms(programs map[int]*gobpfman.ListResponse_Lis
 	}
 }
 
+var currentID = 1000
+
 func (b *BpfmanClientFake) Load(ctx context.Context, in *gobpfman.LoadRequest, opts ...grpc.CallOption) (*gobpfman.LoadResponse, error) {
-	id := rand.Intn(100)
+	currentID++
+	id := currentID
```

Comment:
do u still need currentID ?

Reply:
BTW, I've seen this issue for a long time.  I.e., unit tests would fail sometimes, but when you run them again they would pass.  It just started happening often enough when I started using more programs that it bothered me enough to figure out why.

Reply:
i was just thinking of adding a cntr of 100 or 100 retries to get unused id from random instead of infinite loop

Reply:
I understood what you were suggesting, and I think it would work, but I think this is better -- it's simpler and always unique because you remember the last ID and increment it for the next one.

I thought about doing it this way after I made the first change, but figured the previous way was good enough and didn't bother going back to change it.

Reply:
sure I am fine either way was just answering ur question